### PR TITLE
Added Security Groups Not VPC #709

### LINKED ISSUE #709: Add Security Groups should have VPC query for AWS CouldFormation
### Platform
CouldFormation

### Provider
AWS

### Description
Security Groups must have VPC

### PR DESCRIPTION
**Pull request by joelcarvalhocheckmarx**
Security Groups must have VPC.

Closes #709

### CODE REVIEW

File: assets/queries/cloudFormation/security_groups_not_vpc/metadata.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "id": "security_groups_not_vpc",
+  "queryName": "Security Groups Not VPC",
```
Comment (joaomartinscx) [inline assets/queries/cloudFormation/security_groups_not_vpc/metadata.json:3]: "Security Groups Do Not Have VPC"

File: assets/queries/cloudFormation/security_groups_not_vpc/query.rego [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+                "keyExpectedValue": sprintf("Resources.%s.Properties.VpcId.Ref is defined", [name]),
+                "keyActualValue": 	sprintf("Resources.%s.Properties.VpcId.Ref is undefined", [name])
+              }
+}
+
+CxPolicy [ result ] {
+   document := input.document
+   resources := document[i].Resources[name]
+   default_name := check_group_name(resources)
+   default_name
```
Comment (joaomartinscx) [inline assets/queries/cloudFormation/security_groups_not_vpc/query.rego:22]: Why is a SecurityGroup having "default" as GroupName is a failure?

File: assets/queries/cloudFormation/security_groups_not_vpc/query.rego [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package Cx
+
+CxPolicy [ result ] {
+   document := input.document
+   resources := document[i].Resources[name]
+   not_exists_vpc := check_exists_vpc(resources)
```
Comment (joaomartinscx) [inline assets/queries/cloudFormation/security_groups_not_vpc/query.rego:6]: When a SecurityGroup has the name "default" doesn't need VPC

Comment (joaomartinscx) [inline assets/queries/cloudFormation/security_groups_not_vpc/query.rego:6]: To improve readability, make *check_exists_vpc* return true when vpc **does** exist and apply the not operator outside.

File: assets/queries/cloudFormation/security_groups_not_vpc/query.rego [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+   not_exists_vpc
+   
+      result := {
+                "documentId": 		input.document[i].id,
+                "searchKey":        sprintf("Resources.%s.Properties.VpcId.Ref", [name]),
+                "issueType":		   "IncorrectValue",
```
Comment (joaomartinscx) [inline assets/queries/cloudFormation/security_groups_not_vpc/query.rego:12]: When an attribute is not defined, the issueType should be "MissingAttribute"

**Review by joaomartinscx — CHANGES_REQUESTED** [review]

**Review by joaomartinscx — COMMENTED** [review]
Everything looks good to me, I recommend the approval and merge of this PR

**Review by joelsou5a — APPROVED** [review]

**Pull request merged**